deepcell/applications/multiplex_segmentation.py

```diff
@@ -0,0 +1,175 @@
+                            norm_method='std',
+                            num_semantic_heads=4,
+                            num_semantic_classes=[1, 1, 2, 3],
+                            location=True,
+                            include_top=True)
+
+        if use_pretrained_weights:
+            weights_path = get_file(
+                os.path.basename(WEIGHTS_PATH),
+                WEIGHTS_PATH,
+                cache_subdir='models',
+                md5_hash='685c4df646abab594b8906919fc31e25'
+            )
+
+            model.load_weights(weights_path)
+        else:
+            weights_path = None
+
+        super(MultiplexSegmentation, self).__init__(model,
+                                                    model_image_shape=model_image_shape,
+                                                    model_mpp=2.0,
+                                                    preprocessing_fn=None,
+                                                    postprocessing_fn=deep_watershed_mibi,
+                                                    dataset_metadata=self.dataset_metadata,
+                                                    model_metadata=self.model_metadata)
+
+    def predict(self,
+                image,
+                batch_size=4,
+                image_mpp=None,
+                preprocess_kwargs={},
+                postprocess_kwargs={}):
+        """Generates a labeled image of the input running prediction with
+        appropriate pre and post processing functions.
+
+        Input images are required to have 4 dimensions `[batch, x, y, channel]`. Additional
+        empty dimensions can be added using `np.expand_dims`
+
+        Args:
+            image (np.array): Input image with shape `[batch, x, y, channel]`
+            batch_size (int, optional): Number of images to predict on per batch. Defaults to 4.
+            image_mpp (float, optional): Microns per pixel for the input image. Defaults to None.
+            preprocess_kwargs (dict, optional): Kwargs to pass to preprocessing function.
+                Defaults to {}.
+            postprocess_kwargs (dict, optional): Kwargs to pass to postprocessing function.
+                Defaults to {}.
+
+        Raises:
+            ValueError: Input data must match required rank of the application, calculated as
+                one dimension more (batch dimension) than expected by the model
+
+            ValueError: Input data must match required number of channels of application
+
+        Returns:
+            np.array: Labeled image
+            np.array: Model output
+        """
+        return self._predict_segmentation(image,
+                                          batch_size=batch_size,
+                                          image_mpp=image_mpp,
+                                          preprocess_kwargs=preprocess_kwargs,
+                                          postprocess_kwargs=postprocess_kwargs), \
+            self.output_images
```

Comment:
I'm not sure about returning output_images at all, but this is hard to read regardless.

```suggestion
        prediction = self._predict_segmentation(
            image,
            batch_size=batch_size,
            image_mpp=image_mpp,
            preprocess_kwargs=preprocess_kwargs,
            postprocess_kwargs=postprocess_kwargs)
        return  prediction, self.output_images
```

Reply:
The other option would be not to return the output in predict, and just to store it as self.variable, which can then be accessed directly?

Basically, I want the option to be able to re-run the post-processing function with different parameters on the raw images, since currently this model often requires some manual tweaking. I'm not sure what the best way to get at the underlying model output is, totally open to suggestions

Reply:
I see. Maybe we could add another method to the base `Application` class (something like `_predict`) that does all the tiling/untiling and returns the model output. Then change `_predict_segmentation` to use the new method.

Reply:
For sure. @msschwartz21 and I had discussed maybe adding a multiplex-specific segmentation function. Any thoughts on this? Happy to go with whatever you both think makes the most sense.

Reply:
How would it differ?

Reply:
Ah got it, okay that makes sense. So one function would return the output. A second function would produce the post-processed model output, and would call the first function/check if the model output has already been saved as internal variable?

Reply:
I think the second function can just take the output as an argument and return the results. (it may exist already as `_postprocess` or something). The outer wrapper for `predict` can just call both.  I'm honestly not sure that `_predict_segmentation` should not just be the body of `predict`, but that's kind of another topic.

I don't think we need an attribute for the output data. (but I could be convinced if there's a good argument for it).

Reply:
Yeah, I also wasn't super clear on the benefit of the _predict_segmentation/predict dichotomy. 

Okay, that sounds good, I'll make those changes. As long as there's some way to directly return model output I don't think it needs to be saved as an attribute.

Reply:
I originally pulled `_predict_segmentation` out as a separate function from `predict` to allow future flexibility for models with a different workflow, e.g. tracking.

Reply:
Ah got it, so the tracking app would still call predict, but it wouldn't simply be a case of a different post-processing function, it would be a fundamentally different workflow?

Reply:
I think it's desired that the main interface for the application is `app.predict()`.

Each one can have a different `predict` implementation, but they will likely re-use a bunch of pieces.